packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts

```diff
@@ -636,6 +645,41 @@ class DescopeWc extends BaseDescopeWc {
           {},
         );
 
+        // Try to detect whether the tab is being throttled when running in a mobile browser, specifically on iOS.
+        // We check whether the tab seems to hidden and the polling callback was called much later than expected,
+        // in which case we allow a much shorter timeout for the polling request. The reschedule check ensures
+        // this cannot happen twice consecutively.
+        const throttled =
+          document.hidden &&
```

Comment:
if I understand this code properly, every request, in a hidden document, that will take more than 500ms will be considered as throttled, is that right?

Reply:
Not the request itself, but rather then scheduled callback being called > 500ms after it should have been called. So we schedule the `/next` call to happen 2 seconds from now, and it takes more than 2.5 seconds, then the condition will be true.